lib/services/elasticsearch.js

```diff
@@ -919,429 +857,380 @@ class ElasticSearch extends Service {
-    delete esRequest.type;
+    return this.client.cat.indices({ format: 'json' })
+      .then(({ body }) => {
+        const
+          esIndexes = body
+            .map(({ index }) => index)
+            .filter(esIndex => {
+              const index = this._extractIndex(esIndex);
 
-    return this.client.indices.getMapping(esRequest)
-      .then(result => {
-        let collections = [];
-
-        if (result[request.input.resource.index]) {
-          collections = Object.keys(
-            result[request.input.resource.index].mappings
-          );
-        }
-
-        return {collections: {stored: collections}};
-      })
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
-  }
-
-  /**
-   * Reset all indexes that the users is allowed to delete
-   *
-   * @param {Request} request
-   * @return {Promise}
-   */
-  deleteIndexes(request) {
-    const deletedIndexes = request.input.body.indexes;
+              if (esIndex[0] !== this.indexPrefix || ! indexes.includes(index)) {
+                return false;
+              }
 
-    if (deletedIndexes === undefined || deletedIndexes.length === 0) {
-      return Bluebird.resolve({deleted: []});
-    }
+              deleted.push(index);
 
-    return this.client.indices.delete({index: deletedIndexes})
-      .then(() => ({deleted: deletedIndexes}))
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
-  }
+              return true;
+            }),
+          esRequest = {
+            index: esIndexes
+          };
 
-  /**
-   * List all known indexes
-   *
-   * @returns {Promise}
-   */
-  listIndexes() {
-    return this.client.indices.getMapping()
-      .then(result => ({
-        indexes: Object.keys(result).filter(
-          indexName => indexName !== '' && indexName[0] !== '%')
+        return this.client.indices.delete(esRequest);
+      })
+      .then(() => ({
+        deleted
       }))
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
-   * Create a new index
+   * Deletes an index
    *
-   * @param {Request} request
-   * @returns {Promise}
-   */
-  createIndex(request) {
-    const index = getIndex(request);
-
-    return this.client.indices.create({index})
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
-  }
-
-  /**
-   * Delete an index
+   * @param {String} index - Index name
    *
-   * @param {Request} request
-   * @returns {Promise}
+   * @returns {Promise.<Object>} { }
    */
-  deleteIndex(request) {
-    const index = getIndex(request);
-
-    delete this.settings.autoRefresh[index];
-
-    return this.client.indices.delete({index})
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
-  }
+  deleteIndex (index) {
+    return this.deleteIndexes([index])
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a semicolon at the end of this statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ASemicolon)